src/components/Icons/Icon.tsx

```diff
@@ -55,51 +55,56 @@ export interface IconProps {
     /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
     modifiers?: string[];
     /** Name of the icon */
-    name: IconNames | LogoNames;
+    name?: IconNames | LogoNames;
     /** Icon role */
     role?: string;
     /** Icon title */
     title?: boolean;
 }
 
-export default class Icon extends React.Component<IconProps, {}> {
-    constructor(props: IconProps) {
-        super(props);
-    }
+export default function Icon(props: React.PropsWithChildren<IconProps>) {
+    let {
+        blockName,
+        decorative,
+        className,
+        desc,
+        iconRotation,
+        modifiers,
+        name,
+        role,
+        title,
+        children,
+    } = props;
 
-    render(): JSX.Element {
-        let {
-            blockName,
-            decorative,
-            className,
-            desc,
-            iconRotation,
-            modifiers,
-            name,
-            role,
-            title,
-        } = this.props;
+    let icon_base_class = "icon";
 
-        let icon_base_class = "icon";
+    if (iconRotation) {
+        modifiers.push(iconRotation);
+    }
 
-        if (iconRotation) {
-            modifiers.push(iconRotation);
-        }
+    let iconProps = {
+        className: bem(icon_base_class, modifiers, blockName, [className]),
+        role: decorative ? "img" : role,
+        "aria-hidden": decorative,
+        "aria-labelledby": title ? "title-" + name : undefined,
+        "aria-describedby": desc ? "desc-" + name : undefined,
+        title: title ? `title-${name}` : undefined,
+        description: desc ? `title-${name}` : undefined,
+    };
 
-        let iconProps = {
-            className: bem(icon_base_class, modifiers, blockName, [className]),
-            role: decorative ? "img" : role,
-            "aria-hidden": decorative,
-            "aria-labelledby": title ? "title-" + name : undefined,
-            "aria-describedby": desc ? "desc-" + name : undefined,
-            title: title ? `title-${name}` : undefined,
-            description: desc ? `title-${name}` : undefined,
-        };
+    let svg;
 
-        let svg = allSvgs[name];
+    if (name && children) {
```

Comment:
let's add a test for this (sorry for the earlier comment where I posted without scrolling down orz)

Reply:
+1

Reply:
💯